CRA.1.1

Aldukhov opened the pull request.

Aldukhov committed:
CRA.1.1

kyzinatra commented on src/components/app/app.jsx:
```diff
@@ -1,17 +1,24 @@
+            <>
+             <AppHeader/>
+             <main className={classNames(styles.main, 'mb-10')}>
+             <Ingredients data={data}/>
```
Можно лучше: круто, что вы знаете о библиотеке `classnames`, но вообще, конечно, такую несложную функцию лучше написать самому. Так мы избавимся от лишних зависимостей и заодно попрактикуемся

kyzinatra commented on src/components/app/app.jsx:
```diff
@@ -1,17 +1,24 @@
-  );
+class App extends React.Component {
+    render() {
+        return (
```
Можно лучше: используйте функциональные компоненты. Они более современные и в них доступны самые новые хуки

kyzinatra commented on src/components/appHeader/appHeader.jsx:
```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+
+class AppHeader extends React.Component {
+    render() {
```
По аналогии лучше использовать более простые функции

kyzinatra commented on src/components/appHeader/appHeader.jsx:
```diff
@@ -0,0 +1,49 @@
+                <ul className={styles.navigation__list}>
+                    <li className={classNames(styles.navigation__element,'pt-4','pb-4','pl-5','pr-5','mr-2')}><div className={classNames(styles.icon,'pr-2')}><BurgerIcon/></div><p className={classNames(`text text_type_main-default` )}>Конструктор</p></li>
+                    <li className={classNames(styles.navigation__element,'pt-4','pb-4','pl-5','pr-5')}><div className={classNames(styles.icon,'pr-2')}><ListIcon/></div><p className={classNames(`text text_type_main-default` )}>Лента заказов</p></li>
+                </ul>
```
Надо исправить: Оберните иконку и текст в ссылку в шапке в ссылку `<a href="#">`

Например

`<a href="#" class="......">`

…
 `Конструктор` 

….
`</a>`

В следующих спринтах вместо ссылки будет компонент Link из библиотеки react router, а пока сделайте просто ссылкой

kyzinatra commented on src/components/burger/ingredients/ingredients.jsx:
```diff
@@ -0,0 +1,97 @@
+    function tabmenu() {
+      return (
+        <div style={{ display: 'flex' }}>
+          <Tab value="Булки" active={current === 'Булки'} onClick={() => setCurrent('Булки')}>
```
Можно лучше: Не стоит использовать **in-line** стили. Их сложно дебажить и у них очень высокий приоритет — лучше вынести их в отдельный класс.

kyzinatra commented on src/components/burger/ingredients/ingredients.jsx:
```diff
@@ -0,0 +1,97 @@
+            Булки
+          </Tab>
+          <Tab value="Соусы" active={current === 'Соусы'} onClick={() => setCurrent('Соусы')}>
+            Соусы
```
Можно лучше: будет здорово выглядеть если по клику на таб список будет прокручиваться к нужному месту. Это можно сделать с помощью `element.scrollIntoView({ behavior: "smooth" });`

Подробнее: https://developer.mozilla.org/ru/docs/Web/API/Element/scrollIntoView

kyzinatra commented on src/components/burger/ingredients/ingredients.jsx:
```diff
@@ -0,0 +1,97 @@
+    const [current, setCurrent] = useState('Булки');
+  
+    function tabmenu() {
+      return (
```
Можно лучше: `tabmenu` выглядит как отдельный компонент. Лучше его так и вынести

kyzinatra commented on src/components/burger/ingredients/ingredients.jsx:
```diff
@@ -0,0 +1,97 @@
+
+class FoodItems extends React.Component {
+
+
```
Можно лучше: разные компоненты стоит выносить по разным файлам. Так мы делаем их более читаемыми и упрощаем их поиск

kyzinatra commented on src/components/burger/ingredients/ingredients.jsx:
```diff
@@ -0,0 +1,97 @@
+                <div className={classNames("styles.sauses-block styles.ingridients-blocks mt-10")}>
+                    <h2 className={classNames("text text_type_main-large")}>Соусы</h2>
+                    <section className={classNames(styles.items)}>
+                        {this.renderItemsOfType("sauce")}
```
Можно лучше: тут не нужен `classNames` так как нет никакой конкатенации. Можно просто `<div className={"styles.sauses-block styles.ingridients-blocks mt-10"}>`

kyzinatra commented on src/components/burger/ingredients/ingredients.jsx:
```diff
@@ -0,0 +1,97 @@
+                </div>
+
+                <div className={classNames("styles.sauses-block styles.ingridients-blocks mt-10")}>
+                    <h2 className={classNames("text text_type_main-large")}>Соусы</h2>
```
Надо исправить: тут `styles.sauses-block` воспринимается как строка, а не как свойсто обьекта. Стоит поправить

`classNames(styles["sauses-block"], styles["ingridients-blocks"], "mt-10")`

kyzinatra commented on src/components/burger/ingredients/ingredients.jsx:
```diff
@@ -0,0 +1,97 @@
+                <div className={classNames("styles.buns-block styles.ingridients-blocks")}>
+                    <h2 className={classNames("text text_type_main-large")}>Булки</h2>
+                    <section className={classNames(styles.items)}>
+                        {this.renderItemsOfType('bun')}
```
Можно лучше: если у вас передается 1 строка, то в `classNames` нет смысла. Можно просто написать:

`<section className={styles.items}>`

или 

`<h2 className={"text text_type_main-large"}>Булки</h2>`

kyzinatra commented on package.json:
```diff
@@ -3,21 +3,22 @@
-    "react-dom": "^17.0.1",
-    "react-intersection-observer": "^8.32.5",
     "react-dnd": "^14.0.5",
     "react-dnd-html5-backend": "^14.1.0",
```
Можно лучше: стоит поудалять зависимости которые вам не нужны и которые вы не используете

Aldukhov committed:
Fix components

change to function comp.
add scrollInto effect
create folders for every components
